stories/__tests__/SQFormDialog.stories.test.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import {fireEvent, render, screen, waitFor} from '@testing-library/react';
+import {composeStories} from '@storybook/testing-react';
+
+import userEvent, {specialChars} from '@testing-library/user-event';
+import * as stories from '../SQFormDialog.stories';
+
+const {Default, WithValidation, WithAutoFocus} = composeStories(stories);
+
+window.alert = jest.fn();
+const handleClose = jest.fn();
+const handleSave = jest.fn();
+
+afterEach(() => {
+  window.alert.mockClear();
+  handleClose.mockClear();
+  handleSave.mockClear();
+});
+
+const mockData = {
+  hello: 'howdy'
+};
+
+describe('Tests for Default', () => {
+  it('should render a form dialog with a title', async () => {
+    const dialogTitleValue = 'Test';
+
+    render(
+      <Default
+        isOpen={true}
+        onSave={handleSave}
+        onClose={handleClose}
+        title={dialogTitleValue}
+      />
+    );
+
+    const dialogTitle = screen.getByText(dialogTitleValue);
```

Comment:
You're getting those `wrapped in act(...) blah blah` errors because there's a state change happening inside of the Formik component. I'm _pretty_ sure that state change is the dialog opening because of the prop `isOpen={true}`. Obviously this state change didn't really affect the test because it still passes, but ideally you should ensure that any state changes have completed before making assertions.

Here the error is caused by using `getByText` before the dialog open state change is complete. So what you can do is "wait" for that text to appear, which should mean that the open state change has completed
```js
await waitFor(() => screen.getByText(dialogTitleValue));
const dialogTitle = screen.getByText(dialogTitleValue);

// or even better

const dialogTitle = await screen.findByText(dialogTitleValue);
```
Then the `expect( ... )` assertion that follows won't run until that `findByText` has resolved.

https://testing-library.com/docs/dom-testing-library/api-async#findby-queries

Reply:
amazing. thank you!